src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -1615,103 +1493,187 @@ void CalcSeriesPIU(EnergyPlusData &state,
+                } else {
+                    thisPIU.PriAirMassFlow =
+                        thisPIU.MaxTotAirMassFlow * (state.dataLoopNodes->Node(thisPIU.SecAirInNode).Temp - MixTempNeeded) /
+                        max(SmallTempDiff,
+                            state.dataLoopNodes->Node(thisPIU.SecAirInNode).Temp - state.dataLoopNodes->Node(thisPIU.PriAirInNode).Temp);
+                    thisPIU.PriAirMassFlow = min(max(thisPIU.PriAirMassFlow, PriAirMassFlowMin), PriAirMassFlowMax);
+                }
+                thisPIU.SecAirMassFlow = max(0.0, thisPIU.MaxTotAirMassFlow - thisPIU.PriAirMassFlow);
+            } else if (thisPIU.fanControlType == FanCntrlType::VariableSpeedFan) {
+                CalcVariableSpeedPIUCoolingBehavior(state, PIUNum, ZoneNode, QZnReq, QToHeatSetPt, PriOn, PriAirMassFlowMin, PriAirMassFlowMax);
             }
-            SecAirMassFlow = max(0.0, thisPIU.MaxTotAirMassFlow - PriAirMassFlow);
         }
     } else {
         // unit is off ; no flow
-        PriAirMassFlow = 0.0;
-        SecAirMassFlow = 0.0;
-    }
-    // Set inlet node flowrates
-    state.dataLoopNodes->Node(thisPIU.PriAirInNode).MassFlowRate = PriAirMassFlow;
-    state.dataLoopNodes->Node(thisPIU.SecAirInNode).MassFlowRate = SecAirMassFlow;
+        thisPIU.heatingOperatingMode = HeatOpModeType::HeaterOff;
+        thisPIU.coolingOperatingMode = CoolOpModeType::CoolerOff;
+        thisPIU.PriAirMassFlow = 0.0;
+        thisPIU.SecAirMassFlow = 0.0;
+    }
+    // set inlet node flowrates
+    state.dataLoopNodes->Node(thisPIU.PriAirInNode).MassFlowRate = thisPIU.PriAirMassFlow;
+    state.dataLoopNodes->Node(thisPIU.SecAirInNode).MassFlowRate = thisPIU.SecAirMassFlow;
     if (PriAirMassFlowMax == 0) {
         thisPIU.PriDamperPosition = 0;
     } else {
-        thisPIU.PriDamperPosition = PriAirMassFlow / PriAirMassFlowMax;
+        thisPIU.PriDamperPosition = thisPIU.PriAirMassFlow / PriAirMassFlowMax;
     }
-    // now that inlet airflows have been set, the terminal bos components can be simulated.
 
+    // now that inlet airflows have been set, the terminal components can be simulated.
     // fire the mixer
     SimAirMixer(state, thisPIU.MixerName, thisPIU.Mixer_Num);
+
     // fire the fan
     if (thisPIU.Fan_Num == DataHVACGlobals::FanType_SystemModelObject) {
-        state.dataHVACFan->fanObjs[thisPIU.Fan_Index]->simulate(state, _, _);
+        if (thisPIU.fanControlType == FanCntrlType::VariableSpeedFan) {
+            // calculate fan speed ratio
+            Real64 fanFlowRatio(1.0);
+            if (thisPIU.MaxTotAirMassFlow > 0.0) {
+                fanFlowRatio = (thisPIU.PriAirMassFlow + thisPIU.SecAirMassFlow) / thisPIU.MaxTotAirMassFlow;
+            }
+            state.dataHVACFan->fanObjs[thisPIU.Fan_Index]->simulate(state, fanFlowRatio, _);
+        } else {
+            state.dataHVACFan->fanObjs[thisPIU.Fan_Index]->simulate(state, _, _);
+        }
```

Comment:
This might seem redundant but it is setup this way to avoid regression diffs due to #10438.

Reply:
Since your math looks correct, which fan output report changed in the existing example files? FYI, I just fixed a fan sizing issue in PIU in a forked repo.

Reply:
Here's what I see in the EIO file for `PlantLoopHeatPump_EIR_Large-Office-2-AWHP-DedHR-AuxBoiler-Pri-Sec-HW.idf` when running it from this branch but also from `develop`:
```
 Component Sizing Information, AirTerminal:SingleDuct:SeriesPIU:Reheat, CORE_BOTTOM PIUHEATING, Design Size Maximum Primary Air Flow Rate [m3/s], 4.18760
 Component Sizing Information, Fan:SystemModel, CORE_BOTTOM PIUHEATING FAN, Design Size Design Maximum Air Flow Rate [m3/s], 4.51031
```
The file is intended to be run for Chicago but I'm running it for Golden, CO.

Reply:
If you look at SystemAirFlowSizing.cc you will see that the TU fan sizes using FinalZoneSizing data while the TU sizes using TermUnitFinalZoneSizing. So the TU needs to set a flag to use the correct data for sizing a fan. This is what I set over in PoweredInductionUnits.cc to get fan sizing correct for SeriesPIU:

    state.dataSize->ZoneEqSizing(state.dataSize->CurZoneEqNum).SystemAirFlow = true;
    state.dataSize->ZoneEqSizing(state.dataSize->CurZoneEqNum).AirVolFlow = 
        state.dataPowerInductionUnits->PIU(PIUNum).MaxTotAirVolFlow;

Reply:
Thanks, I'll take a look at it. Is that okay if I include these changes in a dedicated PR for #10438? There's already a lot of changes (in my opinion) in this PR and the CI comes out relatively clean, at least with a manageable number of (expected) diffs.

Reply:
Yes, OK to include this fan sizing change in #10438.